src/analysis/rule_extraction.jl

```diff
@@ -3,26 +3,26 @@ using StableRNGs: StableRNG
 import ..performance: temporal_variability
 
 """
-    Rule{V<:Vector{Float64},W<:Vector{Vector{Any}}}
+    Rule{V<:Vector{Vector},W<:Vector{Float64}}
 
-A Rule contains a condition (vector of conditional clauses) and consequent (vector 
+A Rule contains a condition (vector of conditional clauses) and consequent (vector
 of values for then and else results of the condition)
 """
-struct Rule{V<:Vector{Float64},W<:Vector{Vector{Any}}}
-    condition::W
-    consequent::V
+struct Rule{V<:Vector{Vector},W<:Vector{Float64}}
+    condition::V
```

Comment:
Does this have to be flexible to `Any`? Or could we tighten the expect type to:

`struct Rule{V<:Vector{Vector{StableRules}},W<:Vector{Float64}}`

Reply:
Condition is not a `Vector{Vector{StableRules}}`, it is a Vector of Vectors and each internal Vector has 3 entries: [String, Symbol, Float64]. One possibility is to create the struct:

```julia
struct Clause
feature::String
direction::Symbol  # :L means < and :R means >=
breakpoint::Float64
end
```

And then Condition would be a `Vector{Clause}`. What I like about this approach is that the code I will write accessing this information becomes more readable (e.g., instead of writing `condition[1]` I can write `condition.feature`).

Reply:
Does `direction` need to be a Symbol? What will you do with `:L` / `:R`?

Reply:
I assume also that `feature` needs to be a string to allow for easy print out and concatenation with other strings/rule descriptions?

Just checking as I would prefer `Symbol` if the above is not the case.

Reply:
SIRUS returns a Symbol, and I was just using it. For now I need to use it to

- print either `<` or `>=` symbols 
- when plotting, I do things like: `direction == :L ? [start_point, breakpoint] : [breakpoint, end_point]`
- later, when we actually use this to filter some dataframe, I think we will do something like the above point `direction == :L ? df[:, start:breakpoint] : df[:, breakpoint:end]`

Reply:
About `feature`, I think it would be just as easy if it was a Symbol. I can change that.